### PR TITLE
chore(packaging): configure rollup for jitar

### LINKED ISSUE #249: Configure rollup for jitar

### PR DESCRIPTION
**Pull request by BasMasking**
Fixes #249 

Changes proposed in this pull request:
- Added rollup configuration to bundle jitar in servable packages
- Use the name field for type checking in reflection
- Reroute the external jitar dependencies to the internal jitar dependencies

@MaskingTechnology/jitar

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "root": true,
-    "ignorePatterns": ["**/dist/**/*", "**/node_modules/**/*","**/coverage/**/*", "**/*config*", "packages/create-jitar/templates/*"],
+    "ignorePatterns": ["**/lib/**/*", "**/dist/**/*", "**/node_modules/**/*","**/coverage/**/*", "**/*config*", "packages/create-jitar/templates/*"],
```
Comment (petermasking) [inline .eslintrc.json:3]: Die we still need to igore the `lib` folder?

File: package.json [hunk 2]
```diff
@@ -34,6 +34,11 @@
         "test": "npm run test --workspace=packages/reflection --workspace=packages/serialization --workspace=packages/runtime --workspace=packages/caching --workspace=packages/server-nodejs --workspace=packages/jitar --workspace=packages/create-jitar --workspace=packages/plugin-vite --if-present"
     },
     "devDependencies": {
+        "@rollup/plugin-json": "^6.0.0",
```
Comment (petermasking) [inline package.json:37]: This plugin seems to be unused

File: packages/.gitignore [hunk 3]
```diff
@@ -1,5 +1,7 @@
 coverage
+build
 dist
+lib
```
Comment (petermasking) [inline packages/.gitignore:4]: Do we still need this lib?

File: packages/.gitignore [hunk 4]
```diff
@@ -1,5 +1,7 @@
 coverage
+build
```
Comment (petermasking) [inline packages/.gitignore:2]: Do we still need this build?

File: packages/jitar/tsconfig.json [hunk 5]
```diff
@@ -1,12 +1,11 @@
 {
   "compilerOptions": {
-    "target": "es2022",
-    "experimentalDecorators": true,
-    "emitDecoratorMetadata": true,
-    "module": "es2022",
+    "target": "ESNext",
+    "module": "ESNext",
+    "declaration": true,
+		"declarationDir": "./dist/types",
```
Comment (petermasking) [inline packages/jitar/tsconfig.json:6]: This line seems to have tabs instead of spaces.

File: packages/runtime/package.json [hunk 6]
```diff
@@ -18,7 +18,8 @@
     "release": "npm run clean && npm run build && npm publish"
   },
   "dependencies": {
-    "@jitar/serialization": "^0.4.0"
+    "@jitar/serialization": "^0.4.0",
+    "@jitar/errors": "^0.4.0"
```
Comment (petermasking) [inline packages/runtime/package.json:22]: The error can be removed

File: packages/server-nodejs/package.json [hunk 7]
```diff
@@ -20,11 +20,10 @@
     "dependencies": {
         "@jitar/caching": "^0.4.0",
         "@jitar/runtime": "^0.4.0",
-        "class-transformer": "^0.5.1",
-        "class-validator": "^0.14.0",
+        "@jitar/errors": "^0.4.0",
```
Comment (petermasking) [inline packages/server-nodejs/package.json:23]: Th errors can be removed.

**Review by petermasking — CHANGES_REQUESTED** [review]
Looks great! Needs a last cleanup round.

File: package.json (hunk 2 above)
Comment (BasMasking) [reply package.json:37]: Correct, removed

File: packages/.gitignore (hunk 4 above)
Comment (BasMasking) [reply packages/.gitignore:2]: Not at the moment. We can add it if required later.

File: packages/runtime/package.json (hunk 6 above)
Comment (BasMasking) [reply packages/runtime/package.json:22]: Oops, forgotten

File: packages/server-nodejs/package.json (hunk 7 above)
Comment (BasMasking) [reply packages/server-nodejs/package.json:23]: I made quite a mess of cleaning up ...

**Review by petermasking — APPROVED** [review]

**Pull request merged**